cylc/flow/scripts/validate_reinstall.py

```diff
@@ -92,6 +92,31 @@ def get_option_parser() -> COP:
     return parser
 
 
+def check_tvars_and_workflow_stopped(
+    is_running: bool, tvars: list, tvars_file: list
+) -> bool:
+    """are template variables set and workflow stopped?
+
+    Template vars set by --set (options.templatevars) or --set-file
+    (optiions.templatevars_file) are only valid if the workflow is stopped
+    and vr will play it.
+
+    args:
+        is_running: Is workflow running?
+        tvars: options.tvars, from `--set`
+        tvars_file: options.tvars_file, from `--set-file`
+    """
+    if is_running and (tvars or [] or tvars_file or []):
+        msg = (
+            'Template variables can only be changed if a'
+            ' workflow is stopped. You tried to change:')
+        for tvar in tvars or [] + tvars_file or []:
+            msg += f'\n * {tvar}'
+        LOG.warning(msg)
```

Comment:
I don't think it needs to list the template variables back to the user
```suggestion
        LOG.warning(
            'Template variables (from --set/--set-file) can '
            'only be changed if the workflow is stopped.'
        )
```